cwlVersion: v1.2
class: CommandLineTool
label: "Nanopore Minimap2 Alignment Tool"
doc: |
  This tool prepares Remora data by processing a BAM file (with move table and MD tag)
  through a minimap2-based realignment workflow. It assumes that the BAM file
  was generated by Dorado with --emit-moves and contains the MD tag.
  The command-line is:
  
      samtools fastq -T "*" <in_bam> | minimap2 -y -ax lr:hq <ref_fa> - | samtools view -b -o <out_bam>
  
  The POD5 file is also provided as input for later Remora steps.
  
baseCommand: [bash, -c]
arguments:
  - valueFrom: |
      set -euo pipefail
      echo "Processing BAM file for Remora data preparation"
      samtools fastq -T "*" "$(inputs.in_bam.path)" | \
      minimap2 -y -ax lr:hq "$(inputs.ref_fa.path)" - | \
      samtools view -b -o out.bam
      # Move output to the expected filename:
      mv out.bam "$(inputs.out_bam_basename)"
      echo "Done."
requirements:
  DockerRequirement:
    dockerPull: "ghcr.io/davidroberson/minimap2:250214"
inputs:
  in_bam:
    type: File
    doc: "Input BAM file with move table (--emit-moves) and MD tag."
  pod5:
    type: File
    doc: "Input POD5 file containing signal data."
  ref_fa:
    type: File
    doc: "Reference FASTA file used for alignment."
  out_bam_basename:
    type: string
    default: "aligned.bam"
    doc: "Basename for the output BAM file."
outputs:
  out_bam:
    type: File
    outputBinding:
      glob: $(inputs.out_bam_basename)
  pod5_out:
    type: File
    outputBinding:
      glob: $(inputs.pod5.basename)
